### PR TITLE
feat(Image Styles): Enable consumer image styles

### PR DESCRIPTION
**Pull request by e0ipso**
Adds image styles per consumer.

* [x] Ready for review
* [x] Ready for merge

In order to provide image styles we need to also provide consumers. This PR adds the following tasks:
  * [x] Adds _Consumers_ and _Consumer Image Styles_.
  * [x] Migration for consumers.
  * [x] Default content for all consumers.

### Notes

Since the URIs for our images are external images (for recipes), then image styles are not correctly applied.

However you can see how the image files contain the derivatives for the tutorials. http://127.0.0.1:8888/api/node/tutorial?include=field_image%2Cfield_image.imageFile&fields%5Bnode--tutorial%5D=field_image&fields%5Bimages%5D=imageFile&fields%5Bfiles%5D=url&_consumer_id=0eecd0b9-b149-418f-b5ef-266b9aeeddab

### Proposed Resolution

Add the missing consumers and merge. Then create two follow ups:
  1. Fix the image styles in the recipes.
  1. Gather and create the actual image styles that each consumer will need. Right now we're just using the default ones.

### CODE REVIEW

**Comment by e0ipso** [comment]
@dawehner please review and let me know your feelings about the proposed resolution.

**Comment by yann-yinn** [comment]
edited : wrong issue

**Comment by dawehner** [comment]
I tried this PR out and for some reason the migration didn't run. I'll investigate tomorrow.

File: composer.json [hunk 1]
```diff
@@ -66,8 +66,8 @@
         "drupal/openapi": "^1.0@alpha",
         "drupal/material_admin": "1.x-dev",
         "drupal/snail": "1.x-dev",
-        "drupal/consumers": "1.x-dev",
-        "drupal/consumer_image_styles": "1.x-dev",
+        "drupal/consumers": "^1.0@beta",
+        "drupal/consumer_image_styles": "^1.0@rc",
```
Comment (dawehner) [inline composer.json]: w00t!

File: modules/recipes_magazin_contenta/default_content/consumers.csv [hunk 2]
```diff
@@ -1,2 +1,7 @@
 label,uuid,owner,description,image,image_styles
-Umami Site in Elm,0eecd0b9-b149-418f-b5ef-266b9aeeddab,Contenta team,The Umami Site consumer written in Elm.,elm.png,"medium,large"
+Umami Site Elm,0eecd0b9-b149-418f-b5ef-266b9aeeddab,Contenta team,The Umami Site consumer written in Elm.,elm.png,"recipe_list_350_300,recipe_detail_750_550"
+Umami Site React,8be46df2-4cf7-48f8-ba70-6280ea62a0be,Contenta team,The Umami Site consumer written in React.,reactjs.png,"recipe_list_350_300,recipe_detail_750_550"
```
Comment (dawehner) [inline modules/recipes_magazin_contenta/default_content/consumers.csv]: I hope we don't cause any licensing issues by committing these images?

**Comment by e0ipso** [comment]
Tested in my local successfully.

File: modules/recipes_magazin_contenta/default_content/consumers.csv (hunk 2 above)
Comment (e0ipso) [reply modules/recipes_magazin_contenta/default_content/consumers.csv]: I'm pretty sure that usage is OK. We're not profiting and we're mentioning the frameworks (even promoting them).

**Comment by dawehner** [comment]
Fair. Do you know someone who might know an answer. Maybe Matthew tift?

**Comment by e0ipso** [comment]
> Fair. Do you know someone who might know an answer. Maybe Matthew tift?

No idea.

**Pull request merged**